doc/BGP/BGP-route-aggregation-with-bbr-awareness.md

```diff
@@ -0,0 +1,310 @@
+        }
+    }
+}
+```
+
+#### Config DB Sample
+```json
+{
+    ...
+    "BGP_AGGREGATE_ADDRESS": {
+        "192.168.0.0/24": {
+            "bbr-required": "true",
+            "summary-only": "false"
+        },
+        "fc00::/63": {
+            "bbr-required": "true",
+            "summary-only": "true"
+            "as-set": "true"
+        }
+    }
+    ...
+}
+```
+
+#### Parameters
+We have parameters to control the behavior of aggregate address in bgp container.
+
+##### BBR Required
+It's a boolean to indicate whether only generate aggregated address when BBR feature is enabled.
+
+If it's true and the BBR feature is not enabled, aggregated address won't generated.
+
+##### Summary Only
+It's a boolean to indicate whether only advertise aggregate address only.
+
+If it's true, then details routes will be suppressed and only aggregated address will be advertised.
+
+##### AS Set
+It's a boolean to indicate whether add a as set of details routes in as path when advertising aggregated address.
+
+If it's true, then when advertising aggregated address there will be a as set of detail routes appended at as path.
+
+### State DB Extension
+For every aggregated address, we track its state in state DB, it has two states active and inactive. Active state means the address is configurated in the bgp container, while inactive state means isn't.
+
+#### State DB sample:
+```json
+{
+    ...
+    "BGP_AGGREGATE_ADDRESS": {
+        "192.168.0.0/24": {
+            "state": "inactive"
+        },
+        "fc00::/63": {
+            "state": "active"
+        }
+    }
+    ...
+}
+```
+
+#### State Transition Diagram
+
+<p align=center>
+<img src="img/aggregate_address_state_transition.png" alt="state-transition">
+</p>
+
+
+### Bgp Container Behavior
+The bgp container will subscribe the keys `BGP_AGGREGATE_ADDRESS` and `BGP_BBR` in config DB. The possible events and related process are:
+1. Add address in config DB:
+    - if BRR requirement is satisfied, generate aggregated address in the bgp container and add address in state DB with active state.
+    - else, add address in state DB with inactive state.
+2. Remove address in config DB:
+    - Remove aggregated address in the bgp container and remove address in state DB.
+3. Enable BBR feature in config DB:
+    - In config DB, find out all addresses that has bbr-required equals true and generate aggregated addresses in the bgp container then update state DB with active state.
+4. Disable BBR feature in config DB:
+    - In config DB, find out all addresses that has bbr-required equals true and remove aggregated addresses in the bgp container then update state DB with inactive state.
+5. The bgp container restarted:
+    - First, the bgp container will clean the addresses in state DB and then the bgp container will process all existed config one by one according to 1~4.
+
+To avoid concurrency issue, all operation mentioned above will be put into queue and be processed one by one. 
```

Comment:
Concurrency issue triggered from CLI?

Reply:
Not CLI, but bgpcfgd, currently bgpcfgd will generate and apply commands in batch, however, if those logic of commands generation have any dependence, it may cause concurrency problem, I think this design bare can't trigger that scenario, I will delete this line.